IdentityCore/src/oauth2/MSIDTokenResponse.h

```diff
@@ -30,46 +30,49 @@
 @protocol MSIDRefreshableToken;
 @class MSIDBaseToken;
 
-@interface MSIDTokenResponse : MSIDJsonObject
+@interface MSIDTokenResponse : NSObject <MSIDJsonSerializable>
 
 // Default properties for an openid error response
-@property (readonly) NSString *error;
-@property (readonly) NSString *errorDescription;
-
+@property (nonatomic, nullable) NSString *error;
+@property (nonatomic, nullable) NSString *errorDescription;
 // Default properties for a successful openid response
-@property (readonly) NSInteger expiresIn;
-@property (readonly) NSString *accessToken;
-@property (readonly) NSString *tokenType;
-@property (readonly) NSString *refreshToken;
-@property (readonly) NSString *scope;
-@property (readonly) NSString *state;
-@property (readonly) NSString *idToken;
+@property (nonatomic) NSInteger expiresIn;
+/*!
+ expiresOn isn't part of the spec, but we use it when we need to serialize/deserialize token reponse to/from JSON,
+ because it contains more precise time then expiresIn.
+ */
+@property (nonatomic) NSInteger expiresOn;
```

Comment:
nit: please move this to derived properties section

Reply:
Agreed to keep it here since this is not a derived property.